apps/wallet/src/ui/app/hooks/useGetObject.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type GetObjectDataResponse } from '@mysten/sui.js';
+import {
+    useQuery,
+    type UseQueryResult,
+    useQueryClient,
+} from '@tanstack/react-query';
+
+//TODO use hook useRpc -
+import { api } from '../redux/store/thunk-extras';
+
+export function useGetObject(
+    objectId: string
+): UseQueryResult<GetObjectDataResponse, unknown> {
+    const rpc = api.instance.fullNode;
+    const response = useQuery(
+        ['object', objectId],
+        async () => {
+            return rpc.getObject(objectId);
+        },
+        { enabled: !!objectId }
+    );
+
+    return response;
+}
+
+// Invalidates the cache Objects. called after a transaction
+export function useValidateObjectQueryCache(cacheId: string) {
```

Comment:
not sure about this, I think we should use mutations and as a side effect invalidate cache. This one seems that it will invalidate cache in every render

Reply:
Yeah since it is a hook, I am not sure how mutation will work. I want something on demand. eg after a successful transaction/staking

Reply:
We should use a mutation to do the transaction/staking and there in the mutation onSuccess invalidate the cache.

Reply:
Got it